### PR TITLE
sql: expose module structure

### PR DESCRIPTION
**Pull request by benesch**
Introduce a `plan` module to the `sql` crate, then expose this structure
publically. This gives module documentation a nice place to live.

@sploiselle I'd like to eventually split `statement.rs` and `query.rs` up too, but this seemed like a good first step.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3241)

### CODE REVIEW

File: src/sql/src/plan.rs [hunk 1]
```diff
@@ -0,0 +1,214 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
```
Comment (sploiselle) [inline src/sql/src/plan.rs:9]: Would you be willing to do a 30-second, one-line braindump/kindling for this module-level documentation?

File: src/sql/src/parse.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::ast::Statement;
+
+/// Parses a raw SQL string into a [`Statement`].
```
Comment (sploiselle) [inline src/sql/src/parse.rs:12]: nit: you say `a Statement`, but this returns a `Vec`; is it always `.len()==1`?

Comment (benesch) [reply src/sql/src/parse.rs:12]: No, I'm just a dummy. Thanks.

Comment (sploiselle) [reply src/sql/src/parse.rs:12]: Hardly a dummy!

File: src/sql/src/plan.rs (hunk 1 above)
Comment (benesch) [reply src/sql/src/plan.rs:9]: You bet! Done.

**Pull request merged**